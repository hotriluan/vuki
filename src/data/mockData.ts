// Mock data for testing components

export const mockProducts = [
  {
    id: '1',
    name: 'Nike Air Max 270 React',
    price: 2890000,
    originalPrice: 3200000,
    image: 'https://images.unsplash.com/photo-1542291026-7eec264c27ff?w=400&h=400&fit=crop',
    href: '/products/nike-air-max-270-react',
    badge: 'HOT',
    rating: 4.5,
    reviews: 128,
    category: 'Sneakers',
    brand: 'Nike',
    inStock: true
  },
  {
    id: '2',
    name: 'Adidas Ultraboost 22',
    price: 3200000,
    originalPrice: 3500000,
    image: 'https://images.unsplash.com/photo-1608231387042-66d1773070a5?w=400&h=400&fit=crop',
    href: '/products/adidas-ultraboost-22',
    badge: 'NEW',
    rating: 4.7,
    reviews: 89,
    category: 'Sneakers',
    brand: 'Adidas',
    inStock: true
  },
  {
    id: '3',
    name: 'Timberland 6-Inch Premium Boot',
    price: 4500000,
    originalPrice: 5000000,
    image: 'https://images.unsplash.com/photo-1544966503-7cc5ac882d5f?w=400&h=400&fit=crop',
    href: '/products/timberland-6-inch-premium',
    badge: 'BESTSELLER',
    rating: 4.8,
    reviews: 203,
    category: 'Boots',
    brand: 'Timberland',
    inStock: true
  },
  {
    id: '4',
    name: 'Converse Chuck Taylor All Star',
    price: 1890000,
    image: 'https://images.unsplash.com/photo-1605348532760-6753d2c43329?w=400&h=400&fit=crop',
    href: '/products/converse-chuck-taylor',
    rating: 4.3,
    reviews: 156,
    category: 'Sneakers',
    brand: 'Converse',
    inStock: true
  },
  {
    id: '5',
    name: 'Vans Old Skool',
    price: 2100000,
    image: 'https://images.unsplash.com/photo-1525966222134-fcfa99b8ae77?w=400&h=400&fit=crop',
    href: '/products/vans-old-skool',
    rating: 4.4,
    reviews: 92,
    category: 'Sneakers',
    brand: 'Vans',
    inStock: false
  },
  {
    id: '6',
    name: 'Dr. Martens 1460 Original',
    price: 3800000,
    image: 'https://images.unsplash.com/photo-1582588678413-dbf45f4823e9?w=400&h=400&fit=crop',
    href: '/products/dr-martens-1460',
    rating: 4.6,
    reviews: 174,
    category: 'Boots',
    brand: 'Dr. Martens',
    inStock: true
  }
];

export const mockCategories = [
  {
    id: 'sneakers',
    name: 'Sneakers',
    href: '/category/sneakers',
    icon: '👟',
    description: 'Giày thể thao thời trang',
    productCount: 156,
    image: 'https://images.unsplash.com/photo-1549298916-b41d501d3772?w=600&h=400&fit=crop'
  },
  {
    id: 'boots',
    name: 'Boots',
    href: '/category/boots',
    icon: '🥾',
    description: 'Giày bốt chất lượng cao',
    productCount: 89,
    image: 'https://images.unsplash.com/photo-1544966503-7cc5ac882d5f?w=600&h=400&fit=crop'
  },
  {
    id: 'sandals',
    name: 'Sandals',
    href: '/category/sandals',
    icon: '👡',
    description: 'Dép sandal thoải mái',
    productCount: 67,
    image: 'https://images.unsplash.com/photo-1560769629-975ec94e6a86?w=600&h=400&fit=crop'
  },
  {
    id: 'accessories',
    name: 'Accessories',
    href: '/category/accessories',
    icon: '🎒',
    description: 'Phụ kiện thời trang',
    productCount: 234,
    image: 'https://images.unsplash.com/photo-1553062407-98eeb64c6a62?w=600&h=400&fit=crop'
  }
];

export const mockBrands = [
  {
    id: 'nike',
    name: 'Nike',
    href: '/brand/nike',
    logo: 'https://logos-world.net/wp-content/uploads/2020/04/Nike-Logo.png',
    description: 'Just Do It',
    productCount: 89
  },
  {
    id: 'adidas',
    name: 'Adidas',
    href: '/brand/adidas',
    logo: 'https://logos-world.net/wp-content/uploads/2020/04/Adidas-Logo.png',
    description: 'Impossible is Nothing',
    productCount: 76
  },
  {
    id: 'timberland',
    name: 'Timberland',
    href: '/brand/timberland',
    logo: 'https://logos-world.net/wp-content/uploads/2020/09/Timberland-Logo.png',
    description: 'Built for the Bold',
    productCount: 45
  },
  {
    id: 'converse',
    name: 'Converse',
    href: '/brand/converse',
    logo: 'https://logos-world.net/wp-content/uploads/2020/06/Converse-Logo.png',
    description: 'Create Next',
    productCount: 34
  },
  {
    id: 'vans',
    name: 'Vans',
    href: '/brand/vans',
    logo: 'https://logos-world.net/wp-content/uploads/2020/06/Vans-Logo.png',
    description: 'Off The Wall',
    productCount: 28
  },
  {
    id: 'dr-martens',
    name: 'Dr. Martens',
    href: '/brand/dr-martens',
    logo: 'https://logos-world.net/wp-content/uploads/2020/12/Dr-Martens-Logo.png',
    description: 'Stand for Something',
    productCount: 23
  }
];

export const mockCartItems = [
  {
    id: '1',
    productId: '1',
    name: 'Nike Air Max 270 React',
    price: 2890000,
    originalPrice: 3200000,
    quantity: 1,
    image: 'https://images.unsplash.com/photo-1542291026-7eec264c27ff?w=100&h=100&fit=crop',
    variant: 'Size: 42, Color: Black/White',
    href: '/products/nike-air-max-270-react'
  },
  {
    id: '2',
    productId: '3',
    name: 'Timberland 6-Inch Premium Boot',
    price: 4500000,
    quantity: 2,
    image: 'https://images.unsplash.com/photo-1544966503-7cc5ac882d5f?w=100&h=100&fit=crop',
    variant: 'Size: 41, Color: Wheat',
    href: '/products/timberland-6-inch-premium'
  }
];

export const mockWishlistItems = [
  {
    id: '1',
    productId: '2',
    name: 'Adidas Ultraboost 22',
    price: 3200000,
    originalPrice: 3500000,
    image: 'https://images.unsplash.com/photo-1608231387042-66d1773070a5?w=100&h=100&fit=crop',
    href: '/products/adidas-ultraboost-22',
    addedDate: '2024-10-01'
  },
  {
    id: '2',
    productId: '6',
    name: 'Dr. Martens 1460 Original',
    price: 3800000,
    image: 'https://images.unsplash.com/photo-1582588678413-dbf45f4823e9?w=100&h=100&fit=crop',
    href: '/products/dr-martens-1460',
    addedDate: '2024-09-28'
  }
];

export const mockUserData = {
  isLoggedIn: true,
  user: {
    id: '1',
    name: 'Nguyễn Văn An',
    email: 'nguyenvanan@example.com',
    avatar: 'https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?w=100&h=100&fit=crop&crop=face',
    phone: '0901234567',
    joinDate: '2023-05-15',
    totalOrders: 12,
    totalSpent: 28500000
  }
};

export const mockSearchSuggestions = [
  { id: '1', name: 'Nike Air Max 270', type: 'product' as const, href: '/products/nike-air-max-270', price: 2890000 },
  { id: '2', name: 'Nike Air Force 1', type: 'product' as const, href: '/products/nike-air-force-1', price: 2690000 },
  { id: '3', name: 'Nike React', type: 'product' as const, href: '/products/nike-react', price: 3100000 },
  { id: '4', name: 'Sneakers', type: 'category' as const, href: '/category/sneakers' },
  { id: '5', name: 'Nike', type: 'brand' as const, href: '/brand/nike' },
  { id: '6', name: 'Adidas Ultraboost', type: 'product' as const, href: '/products/adidas-ultraboost', price: 3200000 },
  { id: '7', name: 'Boots', type: 'category' as const, href: '/category/boots' },
];

export const mockMenuNavigation = [
  { 
    name: 'Sneakers', 
    href: '/category/sneakers',
    megaMenu: {
      enabled: true,
      featured: mockProducts.filter(p => p.category === 'Sneakers').slice(0, 2),
      categories: [
        { name: 'Running', href: '/category/sneakers?type=running', icon: '🏃‍♂️' },
        { name: 'Basketball', href: '/category/sneakers?type=basketball', icon: '🏀' },
        { name: 'Lifestyle', href: '/category/sneakers?type=lifestyle', icon: '👟' },
        { name: 'Training', href: '/category/sneakers?type=training', icon: '💪' }
      ],
      brands: mockBrands.filter(b => ['nike', 'adidas', 'converse', 'vans'].includes(b.id))
    }
  },
  { 
    name: 'Boots', 
    href: '/category/boots',
    megaMenu: {
      enabled: true,
      featured: mockProducts.filter(p => p.category === 'Boots').slice(0, 2),
      categories: [
        { name: 'Work Boots', href: '/category/boots?type=work', icon: '👷‍♂️' },
        { name: 'Fashion Boots', href: '/category/boots?type=fashion', icon: '👢' },
        { name: 'Hiking Boots', href: '/category/boots?type=hiking', icon: '🥾' }
      ],
      brands: mockBrands.filter(b => ['timberland', 'dr-martens'].includes(b.id))
    }
  },
  { 
    name: 'Accessories', 
    href: '/category/accessories'
  },
  { 
    name: 'Sale', 
    href: '/sale',
    badge: 'HOT'
  }
];

export const mockRecentlyViewed = [
  mockProducts[0],
  mockProducts[2],
  mockProducts[4]
];

export const mockOrders = [
  {
    id: 'ORD-001',
    date: '2024-09-25',
    status: 'delivered',
    total: 7390000,
    items: [
      { ...mockProducts[0], quantity: 1 },
      { ...mockProducts[2], quantity: 1 }
    ]
  },
  {
    id: 'ORD-002',
    date: '2024-09-15',
    status: 'shipping',
    total: 3200000,
    items: [
      { ...mockProducts[1], quantity: 1 }
    ]
  }
];