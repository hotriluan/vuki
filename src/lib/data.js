// JS mirror of data.ts. Now upgraded to hit the database (Prisma) so that
// accidental resolution to .js (instead of data.ts) still returns fresh data.
// Falls back to bundled JSON only if DB call fails (e.g. during very early build).
import rawProducts from './products.json';
import { prisma } from '@/lib/prisma';

export const categories = [
	{ id: 'c-sneakers', name: 'Sneakers', slug: 'sneakers' },
	{ id: 'c-boots', name: 'Boots', slug: 'boots' },
	{ id: 'c-accessories', name: 'Accessories', slug: 'accessories' },
	{ id: 'c-limited', name: 'Limited', slug: 'limited' }
];

export function getCategories() {
	// Legacy static categories; TS version uses DB
	return categories;
}

// Static fallback snapshot (only used if DB unavailable)
export const products = (rawProducts || []).map(p => ({
	createdAt: new Date().toISOString(),
	...p
}));

export async function findProductBySlug(slug) {
	try {
			return await prisma.product.findFirst({
				where: { slug, deletedAt: null, OR: [ { status: 'PUBLISHED' }, { status: 'SCHEDULED', publishedAt: { lte: new Date() } } ] },
			include: { categories: true, variants: true }
		});
	} catch {
		return products.find(p => p.slug === slug);
	}
}
export async function getProducts() {
	try {
			return await prisma.product.findMany({
				where: { deletedAt: null, OR: [ { status: 'PUBLISHED' }, { status: 'SCHEDULED', publishedAt: { lte: new Date() } } ] },
			include: { categories: true, variants: true }
		});
	} catch {
		return products;
	}
}
export async function findCategoryBySlug(slug) {
	try {
		return await prisma.category.findUnique({ where: { slug } });
	} catch {
		return categories.find(c => c.slug === slug);
	}
}
export async function productsByCategorySlug(slug) {
	try {
		const category = await prisma.category.findUnique({ where: { slug } });
		if (!category) return [];
			return await prisma.product.findMany({
				where: { deletedAt: null, categories: { some: { categoryId: category.id } }, OR: [ { status: 'PUBLISHED' }, { status: 'SCHEDULED', publishedAt: { lte: new Date() } } ] },
			include: { categories: true, variants: true }
		});
	} catch {
		const category = categories.find(c => c.slug === slug);
		if (!category) return [];
		return products.filter(p => (p.categoryIds || []).includes(category.id));
	}
}
export async function findProductById(id) {
	try {
			return await prisma.product.findFirst({
				where: { id, deletedAt: null, OR: [ { status: 'PUBLISHED' }, { status: 'SCHEDULED', publishedAt: { lte: new Date() } } ] },
			include: { categories: true, variants: true }
		});
	} catch {
		return products.find(p => p.id === id);
	}
}
export { products as allProducts };
export default {
	categories,
	getCategories,
	products,
	getProducts,
	findProductBySlug,
	findCategoryBySlug,
	productsByCategorySlug,
	findProductById
};
