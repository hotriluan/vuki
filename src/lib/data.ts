import { Category, Product } from './types';

export const categories: Category[] = [
  { id: 'c-sneakers', name: 'Sneakers', slug: 'sneakers' },
  { id: 'c-boots', name: 'Boots', slug: 'boots' },
  { id: 'c-accessories', name: 'Accessories', slug: 'accessories' },
  { id: 'c-limited', name: 'Limited', slug: 'limited' }
];

export const products: Product[] = [
  {
    id: 'p-1',
    name: 'Urban Runner White',
    slug: 'urban-runner-white',
    description: 'Lightweight everyday sneaker with breathable mesh and cushioned sole.',
    price: 1590000,
    salePrice: 1290000,
    categoryIds: ['c-sneakers'],
  images: ['https://images.unsplash.com/photo-1519744792095-2f2205e87b6f?auto=format&fit=crop&w=800&q=60'],
    featured: true,
    createdAt: new Date().toISOString(),
    variants: [
      { id: 'sz-39', label: 'Size 39', stock: 10 },
      { id: 'sz-40', label: 'Size 40', stock: 8 },
      { id: 'sz-41', label: 'Size 41', stock: 5, priceDiff: 20_000 },
      { id: 'sz-42', label: 'Size 42', stock: 0, priceDiff: 20_000 }
    ]
  },
  {
    id: 'p-2',
    name: 'Trail Explorer',
    slug: 'trail-explorer',
    description: 'Durable outdoor shoe designed for mixed terrain with reinforced toe.',
    price: 1890000,
    salePrice: 1590000,
    categoryIds: ['c-boots'],
    images: ['https://images.unsplash.com/photo-1600185365483-26d7a4cc7519?auto=format&fit=crop&w=800&q=60'],
    createdAt: new Date().toISOString()
  },
  {
    id: 'p-3',
    name: 'Minimalist Leather Boot',
    slug: 'minimalist-leather-boot',
    description: 'Premium full-grain leather boot with clean lines and stitched sole.',
    price: 2350000,
    categoryIds: ['c-boots'],
    images: ['https://images.unsplash.com/photo-1525966222134-fcfa99b8ae77?auto=format&fit=crop&w=800&q=60'],
    featured: true,
    createdAt: new Date().toISOString()
  },
  {
    id: 'p-4',
    name: 'Performance Sock',
    slug: 'performance-sock',
    description: 'Moisture-wicking breathable sock engineered for comfort.',
    price: 99000,
    salePrice: 79000,
    categoryIds: ['c-accessories'],
    images: ['https://images.unsplash.com/photo-1600180758890-c8d3a5d35a5d?auto=format&fit=crop&w=800&q=60'],
    createdAt: new Date().toISOString()
  }
];

export function findProductBySlug(slug: string) {
  return products.find(p => p.slug === slug);
}

export function findCategoryBySlug(slug: string) {
  return categories.find(c => c.slug === slug);
}

export function productsByCategorySlug(slug: string) {
  const category = findCategoryBySlug(slug);
  if (!category) return [];
  return products.filter(p => p.categoryIds.includes(category.id));
}

export function findProductById(id: string) {
  return products.find(p => p.id === id);
}
