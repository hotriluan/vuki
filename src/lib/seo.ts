import type { Product } from './types';

// Base site settings (can evolve to a config file or CMS later)
export const siteName = 'Shoe Store Starter';
export const siteTagline = 'Modern shoe e-commerce starter built with Next.js';

// Resolve site URL from env (no trailing slash)
export function getSiteUrl() {
  const env = process.env.NEXT_PUBLIC_SITE_URL || 'http://localhost:3000';
  return env.replace(/\/$/, '');
}

export function buildCanonical(pathname: string) {
  return getSiteUrl() + pathname;
}

export function formatPriceForMeta(product: Product) {
  // Return raw number (VND) for structured data (no formatting separators)
  const active = product.salePrice && product.salePrice < product.price ? product.salePrice : product.price;
  return active;
}

export function productTitle(product: Product) {
  return `${product.name} | ${siteName}`;
}

export function productDescription(product: Product) {
  // Fallback if description is very long - slice (Google usually uses first ~155 chars)
  const base = product.description || siteTagline;
  return base.length > 155 ? base.slice(0, 152) + '...' : base;
}

export function buildProductMetadata(product: Product) {
  const canonical = buildCanonical(`/product/${product.slug}`);
  const title = productTitle(product);
  const description = productDescription(product);
  const images = product.images?.length ? product.images : undefined;
  return {
    title,
    description,
    alternates: { canonical },
    openGraph: {
      title,
      description,
      url: canonical,
      siteName,
      images: images?.map((url) => ({ url, width: 800, height: 1000, alt: product.name }))
    },
    twitter: {
      card: 'summary_large_image',
      title,
      description,
      images: images?.[0]
    }
  } as const;
}

export function buildProductJsonLd(product: Product) {
  const price = formatPriceForMeta(product) / 1000 * 1000; // price already integer VND; normalize to 1000 step
  const availability = 'https://schema.org/InStock'; // mock inventory assumes in stock; could vary per variant
  return {
    '@context': 'https://schema.org',
    '@type': 'Product',
    name: product.name,
    description: productDescription(product),
    image: product.images,
    sku: product.id,
    offers: {
      '@type': 'Offer',
      priceCurrency: 'VND',
      price,
      availability,
      url: buildCanonical(`/product/${product.slug}`)
    }
  };
}
