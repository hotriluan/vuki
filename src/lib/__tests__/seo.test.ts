import React from 'react';
import { describe, it, expect } from 'vitest';
import { buildProductMetadata, buildProductJsonLd, productDescription } from '../seo';
import { products } from '../data';

const sample = products[0];

describe('seo helpers', () => {
  it('buildProductMetadata contains canonical and twitter image', () => {
    const meta = buildProductMetadata(sample as any);
    expect(meta.alternates?.canonical).toMatch(`/product/${sample.slug}`);
    expect(meta.openGraph?.images?.[0]?.url).toBe(sample.images[0]);
  });

  it('productDescription truncates long descriptions', () => {
    const long = { ...sample, description: 'x'.repeat(300) } as any;
    const desc = productDescription(long);
    expect(desc.length).toBeLessThanOrEqual(155 + 3);
    expect(desc.endsWith('...')).toBe(true);
  });

  it('buildProductJsonLd base shape snapshot', () => {
    const jsonLd = buildProductJsonLd(sample as any);
    expect(jsonLd).toMatchSnapshot();
  });
});
