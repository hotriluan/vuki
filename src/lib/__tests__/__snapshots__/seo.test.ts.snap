// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`seo helpers > buildProductJsonLd base shape snapshot 1`] = `
{
  "@context": "https://schema.org",
  "@type": "Product",
  "description": "Lightweight everyday sneaker with breathable mesh and cushioned sole.",
  "image": [
    "https://images.unsplash.com/photo-1519744792095-2f2205e87b6f?auto=format&fit=crop&w=800&q=60",
  ],
  "name": "Urban Runner White",
  "offers": {
    "@type": "Offer",
    "availability": "https://schema.org/InStock",
    "price": 1290000,
    "priceCurrency": "VND",
    "url": "http://localhost:3000/product/urban-runner-white",
  },
  "sku": "p-1",
}
`;
